### ISSUE TITLE
Install Divest OS on GT-N5100 with latest build 6th of May

### ISSUE THREAD

**Issue by TiberiusRegenschein** [Divested-Mobile/DivestOS-Build#209]
First of all, I appreciate your efforts to give life to old devices like mine with a ROM targeted on security. There is no technical reason to throw my GT-N5100 into the garbage. It is still responsive and the 2nd battery is going strong. Latest firmware updates from Samsung are from end of 2014 and nothing will be done on that end any more, but this is true for all 3rd party ROMs. I am hoping with your ROM to find a solution to use the device a few more years to com - Thank you.

Here what I have done to install Divest OS on the device:
# Preparation
- Installed latest 4.4.4 firmware with Odin, "cleaned" the device and made it full working
  # Flashing GT-N5100 with Heimdall
  More can be found about using heimdall with Ubuntu [here](https://wiki.ubuntuusers.de/Heimdall/)
 ## On your phone¶

 - Open the 'Settings' app
- Navigate to the 'About' page
- Tap on the field labeled 'Build number' 7 times. A toast should appear saying that developer mode has been enabled
- There should now be an screen in the 'Settings' app labeled "Developer options"
- Under "Developer options", enable "Android debugging"
- Shut down the device
- Boot into download mode by pushing volume down and power simultaneously

## On your computer
-  Check connectivity with e.g. `adb devices`
-  If device is recognized
- Download Recovery as .img
- Open terminal in folder with the recovery
-flash recovery with  `heimdall flash --RECOVERY divested ... -dos-n5100-recovery.img --no-reboot` 

Error message: E: failed to find /misc partition

## On your phone
- Boot into recovery by pushing home/volume down / power simultaneously
- Do full factory reset
- `Sideload divest os with adb sideload divested-14.1-date-dos-n5100.zip`
- Do full factory reset
- reboot system
# Working
- System in general
- SD Card
- Bluetooth
- System successfully boots up into Divest OS
- Java lang, Security exception ...
- No SIM, selection of APN not necessary
- WLAN is shown including MAC address, connection not possible. Tested with a HotSpot and a standard router. Message "Password may be incorrect" although password is correct
I created a log for the radio, because that seems to be the most pressing issue. Without network, no update.

Find the log attached
[Radio_log_Divest_OS.log](https://github.com/Divested-Mobile/DivestOS-Build/files/11415601/Radio_log_Divest_OS.log)

**Comment by SkewedZeppelin** [comment]
looks like the modem is looping restart

can you give logs from:
```
adb logcat -b all -d | grep -i audit
adb logcat -b all -d | grep FATAL -c20
```

**Comment by SkewedZeppelin** [comment]
so selinux is the issue here, I'll switch n51xx to -userdebug like I did i9xx series

**Comment by SkewedZeppelin** [comment]
new builds for n51xx series are up, please test

**Comment by TiberiusRegenschein** [comment]
To be on the save side I installed latest TWRP and sideloaded DOVEST OS ROM. No change.

I also installed XDA ROM from here [https://forum.xda-developers.com/t/lineageos-19-1-android-12l-for-gt-n5100-gt-n5110-gt-n5120-beta.4366627/](url) and removed DIVEST OS and everything is back to normal

**Comment by SkewedZeppelin** [comment]
only "works" because they disable selinux which throws security out the window and makes the app sandbox useless
>  Selinux: permissive

**Comment by TiberiusRegenschein** [comment]
You are absolutely right and security is the reason for me to look since a while, even years, now for an alternative to be able to use the device with up to date apps from f-droid. I like the form-factor of 8" and having a pen to scribble. There are not many alternatives out there.
I initiated contact via xmpp:divestos-mobile@conference.konvers.me to keep github, or would you like me to move to gitlab, for issue tracking and solving

**Comment by SkewedZeppelin** [comment]
are you able to test again?

another user recently reported their n5110 working well

**Comment by TiberiusRegenschein** [comment]
Hi, thanks for actively pushing. As of the version September 20th many things are improved:
- The issue with the phone connection was on my end. I needed to flash all parts of the ROM with Odin and then I was able to get a cell signal
- The Wifi problem is odd, because everything works well on the original ROM 4.4.2 from Samsung. I did further tests including a HotSpot from another phone, which was working. Digging deeper into my router settings and playing around with certain settings revealed the following: Setting security to WPA2 + WPA3 (transition mode) is not working, setting the security to WPA2 (CCMP) does work. It looks like the ROM as is has issues on that device with Wifi security WPA2 variants, although everything works fine on the original 4.4.2 ROM

There are some smaller issues still:
- I can download via OTA a new version, but not install it. I get an error message "Updater Installation Error with the triangle warning sign. I can do updates with adb sideload
- Etar, the calendar app is not starting successfully, but other calendar apps do

Confirmed working is:
- GPS
- Encryption

and most of the other stuff as far as I could test. Solving the Wifi issue to not rely on the HotSpot of another phone would be great and would make the device as fresh as it can get after nearly 10 years now.

Thanks again :smiley

**Comment by SkewedZeppelin** [comment]
> I can download via OTA a new version, but not install it. 

This may be a space issue, that device only has 8GB I think? so like 5GB available to use?
The update + extracted size may be too large.

Etar not working is strange though

> WPA2 + WPA3

I've found this is unreliable even on other devices like laptops.
I personally use and recommend keeping a distinct WPA2 and WPA3 SSID if possible.

**Comment by TiberiusRegenschein** [comment]
One update regarding Etar: After installing the latest version from F-Droid Etar works as expected. Looking into ETAR Github there were some issues with the latest version.

Tablet has 16GB of which are 6 GB used. Should not be a space issue. I installed TWRP resulting in the same problem. OTA is working with the same TWRP and other ROMs.

WPA2 + WPA3 is unfortunate but seems to be a problem with other lineage os versions too and your described solution to separate both is also named. => Probably something I have to work around

With this the remaining issue is the OTA update. Any  log I can help with?
